src/NLog/LayoutRenderers/AssemblyVersionLayoutRenderer.cs

```diff
@@ -118,6 +128,24 @@ protected override void Append(StringBuilder builder, LogEventInfo logEvent)
             builder.Append(version);
         }
 
+        private string ApplyFormatToVersion(string version)
+        {
+            var lowerFormat = Format.ToLowerInvariant();
+
+            if (lowerFormat == DefaultFormat || string.IsNullOrEmpty(version))
```

Comment:
for performance reasons, we prefer:

- do the toLowerInvariant in the setter of Format.
- use OrdinalIgnoreCase  comparison. 

Could you please change that? Thanks!